async_service/trio.py

```
@@ -147,7 +184,10 @@ def is_finished(self) -> bool:
     def cancel(self) -> None:
         if not self.is_started:
             raise LifecycleError("Cannot cancel as service which was never started.")
-        self._task_nursery.cancel_scope.cancel()
+        elif not self.is_running:
+            return
```

Do we really want to silently return here? Maybe at least log something as cancelling a service that is not running may indicate a bug lurking somewhere, no?

---

I'm going to open an issue for this.